internal/clients/imagebuilder/image_builder_payloads.go

```diff
@@ -0,0 +1,19 @@
+package imagebuilder
+
+type AmiResponse struct {
+	Distribution  string          `json:"distribution"`
+	ImageName     string          `json:"image_name"`
+	ImageRequests []ImageRequests `json:"image_requests"`
+}
+type Options struct {
+	ShareWithAccounts []string `json:"share_with_accounts"`
+}
+type UploadRequest struct {
+	Type    string  `json:"type"`
+	Options Options `json:"options"`
+}
+type ImageRequests struct {
+	Architecture  string        `json:"architecture"`
+	ImageType     string        `json:"image_type"`
+	UploadRequest UploadRequest `json:"upload_request"`
+}
```

Comment:
You can generate these structs based on the spec file. See https://github.com/osbuild/image-builder/blob/main/internal/composer/client.go#L1 for an example of the magic `// go:generate` comment.  You still need to download the actual spec for that though.

For that I'd recommend to work against the spec currently in production (https://console.redhat.com/api/image-builder/v1/openapi.json). If you pick the production spec (vs stage or upstream), you can upgrade/release as you please as backward compatibility is guaranteed within a major version, but not forward compatibility (relying on new features/endpoints not in production yet).

Reply:
Whatever we can achieve by generation, we should use generation. @adiabramovitch please create Makefile targets for these.